src/sql/src/plan/statement/ddl/connection.rs

```diff
@@ -78,84 +79,96 @@ generate_extracted_config!(
     (Port, u16)
 );
 
-impl ConnectionOptionExtracted {
-    fn ensure_only_valid_options(
-        mut seen: BTreeSet<ConnectionOptionName>,
-        t: CreateConnectionType,
-    ) -> Result<(), PlanError> {
-        use mz_sql_parser::ast::ConnectionOptionName::*;
-
-        let permitted_options = match t {
-            CreateConnectionType::Aws => [
-                AccessKeyId,
-                SecretAccessKey,
-                Token,
-                Endpoint,
-                Region,
-                RoleArn,
-            ]
-            .as_slice(),
-            CreateConnectionType::AwsPrivatelink => &[AvailabilityZones, Port, ServiceName],
-            CreateConnectionType::Csr => &[
-                AwsPrivatelink,
-                Password,
-                Port,
-                SshTunnel,
-                SslCertificate,
-                SslCertificateAuthority,
-                SslKey,
-                Url,
-                User,
-            ],
-            CreateConnectionType::Kafka => &[
-                Broker,
-                Brokers,
-                ProgressTopic,
-                SshTunnel,
-                SslKey,
-                SslCertificate,
-                SslCertificateAuthority,
-                SaslMechanisms,
-                SaslUsername,
-                SaslPassword,
-            ],
-            CreateConnectionType::Postgres => &[
-                AwsPrivatelink,
-                Database,
-                Host,
-                Password,
-                Port,
-                SshTunnel,
-                SslCertificate,
-                SslCertificateAuthority,
-                SslKey,
-                SslMode,
-                User,
-            ],
-            CreateConnectionType::Ssh => &[Host, Port, User],
-        };
+/// Options which cannot be changed using ALTER CONNECTION.
+pub(crate) const INALTERABLE_OPTIONS: &[ConnectionOptionName] = &[ProgressTopic];
+
+/// Options of which only one may be specified.
+pub(crate) const MUTUALLY_EXCLUSIVE_SETS: &[&[ConnectionOptionName]] = &[&[Broker, Brokers]];
+
+pub(super) fn validate_options_per_connection_type(
+    t: CreateConnectionType,
+    mut options: BTreeSet<ConnectionOptionName>,
+) -> Result<(), PlanError> {
+    use mz_sql_parser::ast::ConnectionOptionName::*;
+    let permitted_options = match t {
+        CreateConnectionType::Aws => [
+            AccessKeyId,
+            SecretAccessKey,
+            Token,
+            Endpoint,
+            Region,
+            RoleArn,
+        ]
```

Comment:
(not covered but I think this is just moved code; feel free to ignore)

Reply:
Yeah, this is just code movement